### PR TITLE
[ts] 6/n tests for PaLM Text Model Parser

### PR DESCRIPTION
**Pull request by Ankush-lastmile**
[ts] 6/n tests for PaLM Text Model Parser

## What

tests for PaLM text model parser

Something to note, not resolved, the palm api for node.js and python seem different.

So what should happen in the config json? Best thing to do here is to just standardize it? pr #400 rfc

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/377).
* __->__ #377
* #396
* #397
* #394
* #376
* #375
* #395

### CODE REVIEW

**Review by jonathanlastmileai — COMMENTED** [review]
LGTM but stack needs review

**Review by saqadri — APPROVED** [review]
Approving to unblock, some comments

File: package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (Ankush-lastmile) [reply package.json]: looks like I added types to root of aiconfig, my mistake. Removed

File: typescript/__tests__/parsers/palm-text/palm-config.json
[the remark this answers is not in the document]
Comment (Ankush-lastmile) [reply typescript/__tests__/parsers/palm-text/palm-config.json:1]: updated to suggested name

File: typescript/__tests__/parsers/palm-text/palm-config.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                }
+            }
+        },
+        {
+            "name": "promptWithParams",
+            "input": "What's your favorite building in {{city}}?",
+            "metadata": {
+                "model": {
+                    "name": "models/text-bison-001",
+                    "settings": {}
```
[the remark this answers is not in the document]
Comment (Ankush-lastmile) [reply typescript/__tests__/parsers/palm-text/palm-config.json]: Is this referring to the empty objects? If so removed the empty objects

File: typescript/__tests__/parsers/palm-text/palm.test.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+});
+
+mockGetApiKeyFromEnv.mockReturnValue("test-api-key");
+
+describe("PaLM Text ModelParser", () => {
+  test("serializing params to config prompt", async () => {
+    // no need to instantiate model parser. Load will instantiate it for us since its a default parser
+    const aiConfig = AIConfigRuntime.load(PALM_CONFIG_PATH);
+
+    const completionParams: protos.google.ai.generativelanguage.v1beta2.IGenerateTextRequest = {
```
[the remark this answers is not in the document]
Comment (Ankush-lastmile) [reply typescript/__tests__/parsers/palm-text/palm.test.ts:29]: I couldn't find an exported type for some reason, I expected the same.

File: typescript/lib/parsers/palm.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+/**
+ * Model Parser for PaLM Text Generation. PaLM API Currently does not support streaming
+ * PaLM Text Generation API: https://developers.generativeai.google/models/language#model_variations
+ * Only supports one model: models/text-bison-001
+ */
+
+export class PaLMTextParser extends ParameterizedModelParser {
+  private client: TextServiceClient | null = null;
+  _id = "models/text-bison-001";
```
[the remark this answers is not in the document]
Comment (Ankush-lastmile) [reply typescript/lib/parsers/palm.ts:21]: it sets the id of the Model Parser. Every model parser should have an ID associated with it, for the GPT's we had the parser as a generic and we registered each model with the parser.

PaLM only has 1 model per api so it makes sense to set the id here.

**Comment by Ankush-lastmile** [comment]
Addressed all comments. Going to land this stack and can fix forward any other changes that may come up.

Will creating a video demo/testplan to include

**Pull request merged**